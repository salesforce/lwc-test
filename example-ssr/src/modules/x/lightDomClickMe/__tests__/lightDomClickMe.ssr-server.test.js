const LightDomClickMe = require('../lightDomClickMe').default;
const { renderAndHashComponent } = require('@lwc/jest-ssr-snapshot-utils');

describe('<x-light-dom-click-me>', () => {
    test('should render on the server', async () => {
        const { renderedComponent, snapshotHash } = renderAndHashComponent(
            'x-light-dom-click-me',
            LightDomClickMe
        );
        expect(renderedComponent).toMatchSnapshot(snapshotHash);
    });
});
