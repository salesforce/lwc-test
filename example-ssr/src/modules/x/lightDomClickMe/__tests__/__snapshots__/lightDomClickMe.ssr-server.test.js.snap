// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<x-light-dom-click-me> should render on the server: ecbb2ac8412c9b9fa9c5719b3a3849d1b32a0a4f64d138a6bdf30823d4e85fa5 1`] = `"<x-light-dom-click-me><style type="text/css">.light-dom-content {border: 1px solid #ddd;padding: 20px;background-color: #f9f9f9;text-align: center;}button {margin-top: 10px;padding: 10px 20px;background-color: #0070d2;color: white;border: none;cursor: pointer;font-size: 16px;border-radius: 5px;}button:hover {background-color: #005bb5;}</style><div class="light-dom-content"><h1>Hello from Light DOM!</h1><p>Click the button to change this message!</p><button>Click Me!</button></div></x-light-dom-click-me>"`;
